txjuju/testing/api.py

```diff
@@ -195,20 +197,48 @@ def _formatAnnotationInfo(self, info, verb):
 
     def _formatApplicationInfo(self, info, verb):
         if self.version.startswith("2."):
-            return ["application", verb, {
+            formatted = {
                 "name": info.name,
-                "charm-url": info.charmURL}]
+                "exposed": info.exposed,
+                }
+            if info.charmURL is not None:
+                formatted["charm-url"] = info.charmURL
```

Comment:
Anyways, this a small non-blocking thing, so I'm approving the branch. We can improve this at a later time if we prefer to, and if there's desire to do that.